### PR TITLE
[JENKINS-36716] Spotbugs cleanup

### PR DESCRIPTION
**Pull request by res0nance**
See [JENKINS-36716](https://issues.jenkins-ci.org/browse/JENKINS-36716).
Simple spotbugs cleanup for a dead store and findsecbugs cleanup with using Message.isEqual

### Proposed changelog entries

* No functional change so none
* ...

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/main/java/hudson/model/Slave.java [hunk 1]
```diff
@@ -200,7 +200,6 @@ public Slave(@NonNull String name, String nodeDescription, String remoteFS, int
         getAssignedLabels();    // compute labels now
 
         this.nodeProperties.replaceBy(nodeProperties);
-         Slave node = (Slave) Jenkins.get().getNode(name);
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/model/Slave.java:203]: Any chance it is used for any kind of initialization? Unlikely, but just in case

**Review by oleg-nenashev — APPROVED** [review]
Fine with me if `Jenkins.get().getNode(name);` has no side effects

File: core/src/main/java/hudson/model/Slave.java (hunk 1 above)
Comment (res0nance) [reply core/src/main/java/hudson/model/Slave.java:203]: I checked it and there isn't any.

**Review by StefanSpieker — APPROVED** [review]
Thanks. I should also continue to work on those spotbugs issues ...

File: core/src/main/java/hudson/model/Slave.java (hunk 1 above)
Comment (oleg-nenashev) [reply core/src/main/java/hudson/model/Slave.java:203]: Thanks!

**Comment by oleg-nenashev** [comment]
We may merge it in 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process

**Pull request merged**